test/components/views/dialogs/SpotlightDialog-test.tsx

```diff
@@ -449,4 +449,59 @@ describe("Spotlight Dialog", () => {
             expect(screen.getByText(potatoRoom.name!)).toBeInTheDocument();
         });
     });
+
+    describe("user directory non-matrix attributes", () => {
+        interface IUserExternalAttributesChunkMember extends IUserChunkMember {
+            location?: string;
+        }
+
+        const testPersonExternalAttributtes: IUserExternalAttributesChunkMember = {
+            user_id: "@earthling:matrix.org",
+            display_name: "Earth Person",
+            avatar_url: undefined,
+            location: "Earth", // this attribute could come from a matrix homeserver connected to LDAP
```

Comment:
this test case is not clear: with this, searching by location would require searching for (a substring of) `Earth`, which is also a substring of `earthling` and `Earth Person`. That means it would also match in the current implementation. However with our change, the dialogue should be able to successfully find a user such as
```
            user_id: "@earthling:matrix.org",
            display_name: "Earth Person",
            avatar_url: undefined,
            location: "Mars",
```
with a query of `Mars`.